### PR TITLE
Refactor: JIT & GLES Init

### PR DESCRIPTION
**Pull request by skmp**
I'm thinking of cleaning the source code, module by module, like this.

Also to move to cmake-for-everywhere

Feedback?

PS. libdragoncast sounds good with everyone?

### CODE REVIEW

**Comment by skmp** [comment]
This pull request **introduces 6 alerts** and **fixes 13** when merging d510410a6ac88be1700bcb34bb9293f62def8a4b into 201bf1543616356cb9fa316cfcde7b89dfc09428 - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-97074e91c80c17d09e6c72ec99d2987b0ee9ae74)

**new alerts:**

* 5 for Missing header guard
* 1 for Short global name

**fixed alerts:**

* 6 for Missing header guard
* 3 for Local variable hides global variable
* 2 for Comparison result is always the same
* 1 for Short global name
* 1 for FIXME comment

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
reicast-rend-select

**Comment by skmp** [comment]
This pull request **introduces 6 alerts** and **fixes 13** when merging 79a469475a6b015d9f065b30383e61ad5d26e751 into 201bf1543616356cb9fa316cfcde7b89dfc09428 - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-ec681cf6b6f46037bf7bfff31166bcacbce369e1)

**new alerts:**

* 5 for Missing header guard
* 1 for Short global name

**fixed alerts:**

* 6 for Missing header guard
* 3 for Local variable hides global variable
* 2 for Comparison result is always the same
* 1 for Short global name
* 1 for FIXME comment

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **introduces 6 alerts** and **fixes 13** when merging 4511e937dcd2739631f9e98c58997395c4f0c762 into 201bf1543616356cb9fa316cfcde7b89dfc09428 - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-3c0ede9a4dea3f8df88f62b327d0ad46f86c90be)

**new alerts:**

* 5 for Missing header guard
* 1 for Short global name

**fixed alerts:**

* 6 for Missing header guard
* 3 for Local variable hides global variable
* 2 for Comparison result is always the same
* 1 for Short global name
* 1 for FIXME comment

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Review by gigaherz — APPROVED** [review]
TL;DR

**Comment by skmp** [comment]
This pull request **introduces 6 alerts** and **fixes 13** when merging dcefd12000c0140b41d107aca459edbf124759b0 into 40971b6e85fb8181b24fb7a1dc46dd49545cfa05 - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-a0f9c3675860cf3a5ee0ed7add40d3fd09c37f1a)

**new alerts:**

* 5 for Missing header guard
* 1 for Short global name

**fixed alerts:**

* 6 for Missing header guard
* 3 for Local variable hides global variable
* 2 for Comparison result is always the same
* 1 for Short global name
* 1 for FIXME comment

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*
